projects/smoke_bts7200/src/main.c

```diff
@@ -0,0 +1,80 @@
+// A simple smoke test for BTS7200
+
+// Periodically take reading from user selected channels and log the result
+// Configurable items: wait time, FRONT or REAR power distro selection, channels to be tested
+#include "bts_7200_current_sense.h"
+#include "current_measurement.h"
+#include "current_measurement_config.h"
+
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "mux.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// Smoke test settings. Can be modified to fit testing purpose.
+#define CURRENT_MEASURE_INTERVAL_MS 500  // Set wait time between each set of readings
+#define IS_FRONT_POWER_DISTRO true       // Set whether to test FRONT or REAR power distro
+
+// Number of channels to be tested, MUST match the number of elements in test_channels
+#define NUM_TEST_CHANNELS 8
+// Set of channels to be tested, the number of elements MUST match NUM_TEST_CHANNELS.
+// Range of channels: 0-7
+static uint8_t test_channels[NUM_TEST_CHANNELS] = { 0, 1, 2, 3, 4, 5, 6, 7 };
+
+static Bts7200Storage s_bts7200_storages[NUM_TEST_CHANNELS];
```

Comment:
For the point about `NUM_TEST_CHANNELS` earlier: you can probably define a `MAX_TEST_CHANNELS` constant with the maximum number of distinct channels (probably 8) and use that here instead of `NUM_TEST_CHANNELS`. You'll have a few unused storages if the size of `test_channels` is less than the maximum but that's ok.